### ISSUE TITLE
Display error

### ISSUE THREAD

**Issue by FTno** [forrestguice/SuntimesWidget#207]
At the bottom of the display in the 'Summer Solstice' section, only the upper part of the text under the date is shown (barely) as seen in the screenshot.

![suntimes](https://user-images.githubusercontent.com/16176811/40604848-afd6d524-6260-11e8-9379-9cff9406c3f0.png)

It's not possible to scroll down to see the text.

(Samsung Galaxy S7 Nougat)

**Comment by forrestguice** [comment]
> It's not possible to scroll down to see the text.

Yeah, unfortunately its an issue with the layout and smaller screens (not enough vertical screenspace). The main card view (containing the twilight times / moon) has a scrollview to deal with this, but the other cards do not. They are simply pushed off screen. The problem gets worse when enabling "blue hour" times - that solstice card gets pushed completely off the screen. Its the same situation w/ the landscape orientation except w/ even less vertical space to work with.

A workaround is to disable some of the UI options to gain additional screen space; e.g. disabling golden hour, the lightmap, etc, should allow that solstice card the space it needs.

I'm not really sure how best to solve it. Maybe another scrollview on the main view, but then its a situation of nested scrollviews (and the technical difficulties that go along). I can also imagine a feature allowing the card's to be reordered, making it possible to choose which views get available space.

**Comment by sjvudp** [comment]
Say every component on the view has a specific minimum height required to display. If it is possible to query "how high" it is allowed to be with the current view, and that height is less than needed, it could just render as a "..." button (pressure sensitive area) using up less vertical space.
The ugly part then will be that the user has to hit the "..." to see more (like maximizing/minimizing).
Maybe instead of popping up some new view to display the desired component, the selected component could "expand", while some other component(s) could "collapse" to make available the vertical space needed.
I don't know how hard or easy that will be to implement, however.
